packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -316,10 +315,7 @@ class AttachCommand extends FlutterCommand {
         result = await runner.attach();
         assert(result != null);
       }
-      if (result == 0) {
```

Comment:
As a meta-point (this might have been there from before), have we been splitting our flutter attach analytics volume between [flutterUsage.sendEvent/events](https://analytics.google.com/analytics/web/#/report/app-content-event-events/a67589403w118070018p123509034/_.useg=&_.cusegid=userifo6VhVpS0KSb_NchqA80g&explorer-table.plotKeys=%5B%5D&_r.drilldown=analytics.eventCategory:attach/) and [flutterUsage.sendCommand/screen](https://analytics.google.com/analytics/web/#/report/app-content-pages/a67589403w118070018p123509034/_.useg=&_.cusegid=userifo6VhVpS0KSb_NchqA80g&explorer-table.plotKeys=%5B%5D&_r.drilldown=analytics.contentDescription:attach/)?

If so, we might want to consolidate them in one place and make sure the [go/flutter-dash](https://plx.corp.google.com/scripts2/?sql=%23%0A%23%20Source%20dashboard:%20https:%2F%2Fdashboards.corp.google.com%2F_a6085d00_4d5a_4bf8_b284_281bfaf0b4d6%0A%23%20Changing%20this%20script%20will%20not%20affect%20the%20SQL%20data%20source%20of%20the%20Plx%20Dashboard.%0A%23%0Aselect%20date,%20%0A%09case%20when%20screen_name%20like%20%22run%25%22%20then%20%22run%22%20else%20screen_name%20end%20as%20command,%0A%20%20%20%20sum(screen_views)%20as%20count%0Afrom%20jayounglee.ga.tool_eventscreens%0Awhere%20%0A%09screen_name%20like%20%22run%25%22%20or%20%0A%09screen_name%20like%20%22create%25%22%20or%20%0A%09screen_name%20like%20%22attach%25%22%0Agroup%20by%20date,%20command) graph reflects it.

Reply:
See https://github.com/flutter/flutter/pull/33458#discussion_r289952795

Reply:
I saw the comment. I agree that given the flutterUsage.sendEvent/events mechanism, it makes sense to add global annotations to all of them. I was asking about the semantics of FlutterCommands such as flutter attach that both send flutterUsage.sendCommand/screen in FlutterCommand.verifyThenRunCommand and a flutterUsage.sendEvent/events in FlutterCommand.run. 

There was a bit of an existing pattern already going on where events are being used as smaller inner acts such as hot restarting and screens being used for bigger outer flutter CLI acts such as run, attach etc. 

1- should we keep that semantic separated, and
2- should we also log success/failure in flutterUsage.sendCommand/screens as well?

Reply:
Sorry, in my comment in the other PR, I missed the `sendCommand` call in `verifyThenRunCommand`. If it makes more sense to add a success/failure parameter to the `sendCommand` there, then you could put the `runCommand()` in a `try {} except {} finally {}` with the `sendCommand` in the finally. Similar pattern as above in `run()`.

Reply:
This is now centralized. PTAL